ops/hookcmds/_other.py

```diff
@@ -0,0 +1,188 @@
+
+    For more details, see:
+    `Juju | Hook commands | config-get <https://documentation.ubuntu.com/juju/3.6/reference/hook-command/list-of-hook-commands/config-get/>`_
+
+    Args:
+        key: The configuration option to retrieve.
+    """
+    args = ['--format=json']
+    if key:
+        args.append(key)
+    stdout = run('config-get', *args)
+    if key:
+        result = cast('bool | int | float | str', json.loads(stdout))
+    else:
+        result = cast('dict[str, bool | int | float | str]', json.loads(stdout))
+    return result
+
+
+def credential_get() -> CloudSpec:
+    """Access cloud credentials.
+
+    For more details, see:
+    `Juju | Hook commands | credential-get <https://documentation.ubuntu.com/juju/3.6/reference/hook-command/list-of-hook-commands/credential-get/>`_
+    """
+    stdout = run('credential-get', '--format=json')
+    result = cast('dict[str, Any]', json.loads(stdout))
+    return CloudSpec._from_dict(result)
+
+
+def goal_state() -> GoalState:
+    """Print the status of the charm's peers and related units.
+
+    For more details, see:
+    `Juju | Hook commands | goal-state <https://documentation.ubuntu.com/juju/3.6/reference/hook-command/list-of-hook-commands/goal-state/>`_
+    """
+    stdout = run('goal-state', '--format=json')
+    result = cast('GoalStateDict', json.loads(stdout))
+    return GoalState._from_dict(result)
+
+
+def is_leader() -> bool:
+    """Obtain the current leadership status for the unit the charm code is executing on.
+
+    The value is not cached. It is accurate for 30s from the time the method is
+    successfully called.
+
+    For more details, see:
+    `Juju | Hook commands | is-leader <https://documentation.ubuntu.com/juju/3.6/reference/hook-command/list-of-hook-commands/is-leader/>`_
+    """
+    stdout = run('is-leader', '--format=json')
+    result = cast('bool', json.loads(stdout))
+    return result
+
+
+def juju_log(
+    message: str, *, level: Literal['TRACE', 'DEBUG', 'INFO', 'WARNING', 'ERROR'] = 'INFO'
+):
+    """Write a message to the juju log.
+
+    For more details, see:
+    `Juju | Hook commands | juju-log <https://documentation.ubuntu.com/juju/3.6/reference/hook-command/list-of-hook-commands/juju-log/>`_
+
+    Args:
+        message: The message to log.
+        level: Send the message at the given level.
+    """
+    run('juju-log', '--log-level', level, message)
+
+
+def juju_reboot(*, now: bool = False):
+    """Reboot the host machine.
+
+    For more details, see:
+    `Juju | Hook commands | juju-reboot <https://documentation.ubuntu.com/juju/3.6/reference/hook-command/list-of-hook-commands/juju-reboot/>`_
+
+    Args:
+        now: Reboot immediately, killing the invoking process.
+    """
+    if now:
+        run('juju-reboot', '--now')
+        # Juju will kill this process, but to avoid races we force that to be the case.
+        sys.exit()
+        return  # Make it simpler to mock out sys.exit() in tests.
```

Comment:
Maybe we should not be smart here.

When ops uses hooktools, I think that sys.exit() automation should be on the ops side, not hook tools.